src/Microsoft.DotNet.Build.CloudTestTasks/WriteItemsToJson.cs

```diff
@@ -73,6 +73,37 @@ public override bool Execute()
 
                                 jsonWriter.WriteEndArray();
                             }
+                            // if the value is surrounded in square brackets it's meant to be an object.
+                            // split the value into its respective chunks and write it into a JSON object.
+                            else if (mdValue.Length > 0 && mdValue[0] == '{' && mdValue[mdValue.Length - 1] == '}')
+                            {
+                                mdValue = mdValue.Substring(1, mdValue.Length - 2);
+                                jsonWriter.WriteStartObject();
+
+                                var parts = mdValue.Split(new char[] { ';' }, StringSplitOptions.RemoveEmptyEntries);
```

Comment:
I would much prefer we just use the normal JSON format and parse that. I don't understand why we need a new serialization format.
<Properties>{"architecture": "$(Platform)", "configuration": "$(Configuration)"}</Properties>
Seems like it should work fine, unless I'm missing something.